### PR TITLE
[google_maps_flutter] Add a bitmap descriptor that is aware of scale

### PR DESCRIPTION
**Pull request by iskakaushik**
## Description

This makes it so you can create BitmapDescriptors that are aware of screen dpi

## Related Issues

- https://github.com/flutter/flutter/issues/27393

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by iskakaushik** [comment]
We do not need to scale images in android because `Bitmap` doesn't scale images by default

File: packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m [hunk 1]
```diff
@@ -153,6 +153,19 @@ static void InterpretInfoWindow(id<FLTGoogleMapMarkerOptionsSink> sink, NSDictio
   }
 }
 
+static UIImage* scaleImage(UIImage* image, NSNumber* scaleParam) {
+  double scale = 1.0;
+  if (scaleParam != nil && ![scaleParam isKindOfClass:[NSNull class]]) {
```
Comment (cyanglaz) [inline packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m]:
```suggestion
  if ([scaleParam isKindOfClass:[NSNumber class]]) {
```

**Comment by iskakaushik** [comment]
@cyanglaz what use case were you thinking of?

Also, we can always factor it out to a different plugin if need be. For now, google maps is the only consumer of it, so might as well leave it here.

**Comment by JRR-OSU** [comment]
@iskakaushik 

I am not able to get the Size param to do anything when creating an ImageConfiguration, or modifying the default one shown in the example. On Android my 64x64 marker is still really small. On both platforms modifying Size does nothing. Is there something I'm missing?

**Pull request merged**